src/hooks/useSingleCampaignHandler.ts

```diff
@@ -73,6 +78,43 @@ export const useSingleCampaignHandler = () => {
     }
   }, [DAOSEEDER_FACTORY_ADDRESS, id, provider]);
 
+  useEffect(() => {
+    const fetchStageData = async () => {
+      if (campaign && campaign.stageCount > 0 && DAOSEEDER_FACTORY_ADDRESS) {
+        const contract = await getSmartContractWithProvider(
+          DAOSEEDER_FACTORY_ADDRESS,
+          provider,
+          JSON.stringify(DaoSeederFactory.abi)
+        );
+        const obj: ICampaignStage[] = [];
+        for (let i = 0; i < campaign.stageCount; i++) {
+          const stageKey = await contract.getKey(campaign.tokenAddress, i);
```

Comment:
This should be calculated using react. Please seach for "calcGammaPoolAddress" in the gs code.